pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go

```diff
@@ -252,6 +326,17 @@ func (sip *streamIngestionProcessor) merge(
 						return nil
 					}
 
+					// Check for a context cancellation first, so that we can stop
+					// ingesting as soon as possible.
+					// TODO(during review): Does this buy us anything? From my logging
```

Comment:
I think that it's rare that we'll hit this since we'd usually be blocked sending on the `merged` ch anyway when the context is cancelled. I think if we're not seeing any difference when stressing we can remove this.